pkg/apis/core/types.go

```diff
@@ -2187,6 +2187,24 @@ type Handler struct {
 	// TODO: implement a realistic TCP lifecycle hook
 	// +optional
 	TCPSocket *TCPSocketAction
+
+	// GRPC specifies an action involving a GRPC port.
+	// +optional
+	GRPC *GRPCAction
+}
+
+type GRPCAction struct {
+	// Port number of the gRPC service. Number must be in the range 1 to 65535.
```

Comment:
since it's explicitly inconsistent with other port definitions in this file, please call it out as a Note in the comments.